### ISSUE TITLE
Overlap might fail if seqlevels do not match

### ISSUE THREAD

**Issue by alexg9010** [BIMSBbioinfo/pigx_chipseq#37]
https://github.com/BIMSBbioinfo/pigx_chipseq/blob/407ac9b8ef4e69e6c7483eea57f2bfb6103c7a4e/scripts/Functions_Helper.R#L123

**Comment by Blosberg** [comment]
From the BS pipeline, I have the following rule:

```
rule tabulate_seqlengths:
    input:
        rules.bismark_genome_preparation.output
    output:
        seqlengths = DIR_mapped+"Refgen_"+ASSEMBLY+"_chromlengths.csv",
    params:
        chromlines = " | grep Sequence ",
        chromcols  = " | cut -f2,3     ",
        seqnames   = " | sed \"s/_CT_converted//g\" "
    message: fmt("Tabulating chromosome lengths in genome: {ASSEMBLY} for later reference.")
    shell:
        nice('bowtie2-inspect', ['-s ' + GENOMEPATH + "Bisulfite_Genome/CT_conversion/BS_CT", '{params.chromlines}', '{params.chromcols}', '{params.seqnames}', ' > {output}'])
```

It creates a 2-column text file with seqlevels and chromosome lengths. e.g. for the mm10 genome the output looks like this:

```
$ more Refgen_mm10_canon_chromlengths.csv
chr1	195471971
chr2	182113224
chr3	160039680
chr4	156508116
...
chrX	171031299
chrY	91744698
```

and then I pass this file as `chrom_seqlengths` into the Rmd rule like this:

```
seqdat_temp    = read.csv(paste0(out_dir,"/",chrom_seqlengths),
                          sep="\t", header=FALSE, stringsAsFactors=FALSE)
chr.len        = seqdat_temp[,2]
names(chr.len) = seqdat_temp[,1]

myseqinfo = Seqinfo(names(chr.len), seqlengths=chr.len, genome=assembly)
myseqinfo.st = keepStandardChromosomes(myseqinfo)
```

I'm struggling a bit to figure out exactly where in the code this should go, but the basic idea should work If we just add the above file as an input to whatever is using the seqlengths. Does that work?

**Comment by alexg9010** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [pigx-chipseq bug fixes needed before F1000](https://trello.com/c/uZVSztg1/12-pigx-chipseq-bug-fixes-needed-before-f1000)

**Comment by alexg9010** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [pigx-chipseq: check for matching sequence names in genome and annotation](https://trello.com/c/lnlpyokx/14-pigx-chipseq-check-for-matching-sequence-names-in-genome-and-annotation)

**Comment by alexg9010** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Overlap might fail if seqlevels do not match](https://trello.com/c/r00IIysl/54-overlap-might-fail-if-seqlevels-do-not-match)

**Comment by alexg9010** [comment]
should be solved with c5225023258e99776ef34b60665df8c633e447a0
